src/ahbicht/expressions/expression_resolver.py

```diff
@@ -122,3 +135,65 @@ async def _package_async(self, package_key_token: Token) -> Tree[Token]:
         # the package_expression is not None because that's the definition of "has been resolved successfully"
         tree_result = parse_condition_expression_to_tree(resolved_package.package_expression)  # type:ignore[arg-type]
         return tree_result
+
+
+# pylint: disable=no-self-use, invalid-name
+class TimeConditionTransformer(Transformer):
+    """
+    The TimeConditionEvaluator replaces "time conditions" (aka "UB1", "UB2", "UB3") with evaluatable format constraints.
+    This is not what BDEW suggests us to do in the "Allgemeine Festlegungen". BDEW says that "UBx" conditions have to be
+    expanded just like packages: For example "UB1" shall be expanded to "([931] ∧ [932] [490]) ⊻ ([931] ∧ [933] [491])".
```

Comment:
rant may be wrong. wegen xor

Reply:
ich glaube es ist immer noch gut so :) denn selbst mit xor könnte es ja unfulfilled werden.